config: SEGGER_RTT_Conf: remove mutex locking

JordanYates opened the pull request:
Remove mutex locking in favour of the standard IRQ locking mechanism. The primary problem with the mutex implementation is that mutex locking is forbidden in ISR's. This means that any logging from an interrupt context (e.g. LOG_PANIC in an exception handler), will itself trigger another assertion due its attempt to use a mutex.

Furthermore, mutexes are a relatively heavyweight locking scheme, which doesn't necessarily make sense in the context of extremely short locking periods that would be expected from RTT.

This change aligns Zephyr with the default RTT locking scheme, which uses interrupt masking to perform access control.

henrikbrixandersen commented on Config/SEGGER_RTT_Conf.h:
There seems to be some unrelated whitespace changes in this commit?

JordanYates replied:
As noted in the PR description, I cannot for the life of me figure out what they are or where they come from.
They don't show up in my local editor and Github isn't being useful.
Normally I would assume newline ending changes, but as far as I can tell that isn't true.

JordanYates replied:
The file apparently previously had a mix of `\r\n` (mostly) and `\n` line endings (the diff that shows) and vscode decided to fix those without showing that there was a difference.

henrikbrixandersen replied:
Okay, that makes sense. We can of course fix that, but in a separate commit.

JordanYates committed:
config: SEGGER_RTT_Conf: remove mutex locking

Remove mutex locking in favour of the standard IRQ locking mechanism.
The primary problem with the mutex implementation is that mutex locking
is forbidden in ISR's. This means that any logging from an interrupt
context (e.g. LOG_PANIC in an exception handler), will itself trigger
another assertion due its attempt to use a mutex.

Furthermore, mutexes are a relatively heavyweight locking scheme, which
doesn't necessarily make sense in the context of extremely short locking
periods that would be expected from RTT.

This change aligns Zephyr with the default RTT locking scheme, which
uses interrupt masking to perform access control.

Signed-off-by: Jordan Yates <jordan@embeint.com>
